protocols/floodsub/src/protocol.rs

```diff
@@ -132,25 +149,26 @@ where
     TSocket: AsyncWrite + AsyncRead + Send + Unpin + 'static,
 {
     type Output = ();
-    type Error = io::Error;
+    type Error = EncodeError;
     type Future = Pin<Box<dyn Future<Output = Result<Self::Output, Self::Error>> + Send>>;
 
-    fn upgrade_outbound(self, mut socket: TSocket, _: Self::Info) -> Self::Future {
+    fn upgrade_outbound(self, socket: TSocket, _: Self::Info) -> Self::Future {
         Box::pin(async move {
-            let bytes = self.into_bytes();
-
-            upgrade::write_length_prefixed(&mut socket, bytes).await?;
-            socket.close().await?;
-
+            let mut framed = Framed::new(
+                socket,
+                prost_codec::Codec::<rpc_proto::Rpc>::new(MAX_MESSAGE_LEN_BYTES),
+            );
+            let _ = framed.send(self.into_rpc()).await;
+            let _ = framed.close().await;
```

Comment:
```suggestion
            framed.send(self.into_rpc()).await?;
            framed.close().await?;
```

We should not ignore those errors :)

Reply:
Ahh! This just solved a mystery I ran into with the error structs/enums. I am refactoring some stuff :smile:

Reply:
My confusion: (In case I look at this PR again in the future)  the error message that adding the `?` generated was not clear to me. What it was asking was for the `Error` type to handle the output of the `?` and what I had there originally (just the struct) did not handle the prost_code:Error error types. So of course that error message now makes sense. The `type Error` needs be aware of any errors that could come out of the impl.